pkgs/applications/networking/p2p/qbittorrent/default.nix

```diff
@@ -1,58 +1,91 @@
-{ mkDerivation, lib, stdenv, fetchFromGitHub, pkg-config
-, boost, libtorrent-rasterbar, qtbase, qttools, qtsvg
-, debugSupport ? false
-, guiSupport ? true, dbus ? null # GUI (disable to run headless)
-, webuiSupport ? true # WebUI
-, trackerSearch ? true, python3 ? null
+{ lib
+, stdenv
+, fetchFromGitHub
+
+, boost
+, cmake
+, Cocoa
+, libtorrent-rasterbar
+, ninja
+, qtbase
+, qtsvg
+, qttools
+, wrapQtAppsHook
+
+, guiSupport ? true
+, dbus ? null
+, qtwayland
+
+, trackerSearch ? true
+, python3 ? null
```

Comment:
surely py3 is available everywhere?

Reply:
Your point? This piece of code and an assert below ensures the `python3` input isn't `null` when `trackerSearch` is `true`. If `trackerSearch` is `false`, there is no need for `python3`.

Reply:
the ` ? null` is superfluous since `callPackage` will ensure it's always provided as py3 is available on all systems. You can still call it with `python3 = null;` if you want (I'm not sure why, but that's a different issue).

Reply:
Ah, I see what you mean. All right.